### DISCUSSION TITLE
Reading query parameters acts differently in dev preview and production build in Next13

### DISCUSSION THREAD

**Discussion by matt1as** [vercel/next.js#45554] [Help]
Hello, 

I'm new to NextJS/React, and trying to get my head around how to read URL Query parameters in a client component. 
If I run the app locally using `npm run dev` everything works as expected, but if I make a production build and run using it using `npm start`, the searchParameters variable is undefined.

Regards
This is the relevant code I have
```
'use client'
import Image from 'next/image'
import { Inter } from '@next/font/google'
import styles from '@/styles/Home.module.css'
import Link from 'next/link'
import useSWR from 'swr'

const inter = Inter({ subsets: ['latin'] })
const fetcher = (url: RequestInfo | URL) => fetch(url).then((res) => res.json());

export interface NameProps {
  // params: { name: string }; // cannot require should be optional
  params?: { name: string };
  // searchParams: any; // cannot require should be optional
  searchParams?: any;
}

const Home: React.FC<NameProps> =( {params, searchParams} ) => {
  console.log(searchParams)
.....
do something
....}
```

**Comment by dluan** [comment]
Have you tried using `use()`?
It is required for client components. https://beta.nextjs.org/docs/data-fetching/fetching
